datadog_lambda/tracing.py

```diff
@@ -384,12 +384,23 @@ def _parse_high_64_bits(trace_tags: str) -> str:
 
 
 def _generate_sfn_parent_id(context: dict) -> int:
+    """
+    The upstream Step Function can propagate its execution context to downstream Lambdas. The
+    Lambda can use these details to share the same traceID and infer its parent's spanID.
+
+    Excluding redriveCount when its 0 to account for cases where customers are using an old
+    version of the Lambda layer that doesn't use this value for its parentID generation.
+    """
     execution_id = context.get("Execution").get("Id")
+    redrive_count = context.get("Execution").get("RedriveCount")
     state_name = context.get("State").get("Name")
     state_entered_time = context.get("State").get("EnteredTime")
 
+    redrive_postfix = "" if redrive_count == 0 else f"#{redrive_count}"
```

Comment:
Is it possible that redrive _count is null?

Reply:
`Execution.RedriveCount` will **always** be in the SFNs context object, defaulting to 0

Only `Execution.RedriveTime` has a risk of being missing, only appearing when redrive count is 1 or more

https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html

Reply:
Even though this is supposed to be true, I found some instances in customer logs where RedriveCount is null. Wasn't able to reproduce it myself. I'll add some null-check here and also for the node layer change accordingly